src/GLideNUI/GLideNUI.cpp

```diff
@@ -29,27 +29,34 @@ int openConfigDialog(const wchar_t * _strFileName, const char * _romName, bool &
 	if (config.generalEmulation.enableCustomSettings != 0 && _romName != nullptr && strlen(_romName) != 0)
 		loadCustomRomSettings(strIniFileName, _romName);
 
-#ifndef M64P_GLIDENUI
-	int argc = 0;
-	char * argv = 0;
-	QApplication a(argc, &argv);
+	QCoreApplication* pApp = QCoreApplication::instance();
+	bool hasApp = pApp != nullptr;
 
-	QTranslator translator;
-	if (translator.load(getTranslationFile(), strIniFileName))
-		a.installTranslator(&translator);
-#endif // M64P_GLIDENUI
+	if (!hasApp) {
+		int argc = 0;
+		char * argv = 0;
+		pApp = new QApplication(argc, &argv);
+
+		QTranslator translator;
+		if (translator.load(getTranslationFile(), strIniFileName))
+			pApp->installTranslator(&translator);
+	}
 
 	ConfigDialog w(Q_NULLPTR, Qt::WindowTitleHint | Qt::WindowSystemMenuHint | Qt::WindowCloseButtonHint);
 
 	w.setIniPath(strIniFileName);
 	w.setRomName(_romName);
 	w.setTitle();
-#ifndef M64P_GLIDENUI
 	w.show();
-	const int res = a.exec();
-#else
-	const int res = w.exec();
-#endif // M64P_GLIDENUI
+
+	int res;
+	if (!hasApp) {
+		res = pApp->exec();
```

Comment:
Do I get it correct that you are using QCoreApplication::instance() to find if plugin is in zilmar-spec mode or in mupen64plus one? If QCoreApplication::instance() returns valid pointer, the plugin is in M64P_GLIDENUI mode and thus w.exec() should be called?

Reply:
well, I'm using `QCoreApplication::instance()` to detect if the GUI (Zilmar spec or mupen spec) is Qt or not, if it is we can't create another `QApplication`